### PR TITLE
Doc: update documentation on CMake+CUDA

### PR DESCRIPTION
**Pull request by mic84**
## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [x] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

File: Docs/sphinx_documentation/source/GPU.rst [hunk 1]
```diff
@@ -182,15 +182,61 @@ can run it and that will generate results like:
...
+.. raw:: latex
+
+   \end{center}
+
+The target architecture to build for can be specified via the configuration option
 ``-DCUDA_ARCH=<target-architecture>``, where ``<target-architecture>`` can be either
-the name of the NVIDIA GPU, i.e. ``Turing``, ``Volta``, ``Pascal``, ``...`` , or its
-version number, i.e. ``10.0``, ``9.0``, ``8.0``, ``...`` .
+the name of the NVIDIA GPU, i.e. ``Turing``, ``Volta``, ``...`` , or its
+version number, i.e. ``10.0``, ``9.0``,  ``...`` .
```
Comment (ax3l) [inline Docs/sphinx_documentation/source/GPU.rst]:
```suggestion
the name of the NVIDIA GPU generation, i.e. ``Turing``, ``Volta``, ``Ampere``, ``...`` , or its
`compute capability <https://developer.nvidia.com/cuda-gpus>`_, i.e. ``10.0``, ``9.0``,  ``...`` .
```

File: Docs/sphinx_documentation/source/GPU.rst [hunk 2]
```diff
@@ -200,28 +246,60 @@ For example, on Cori GPUs you can specify the architecture as follows:
    cmake [options] -DENABLE_CUDA=yes -DCUDA_ARCH=Volta /path/to/amrex/source
 
 
-Note that AMReX only supports GPU architectures with version number ``6.0`` or higher.
+If no architecture is specified, CMake will try to determine which GPU architecture is
+supported by the system.
+If more than one is found, CMake will build for all of them.
+This will generally results in a larger library and longer build times.
+If autodetection fails, a set of "common" architectures is assumed.
+**Note that AMReX only supports GPU architectures with version number ``6.0`` or higher**.
```
Comment (ax3l) [inline Docs/sphinx_documentation/source/GPU.rst]: I think we also just support CUDA 9.0 and newer. Let's add this as well.

**Review by ax3l — COMMENTED** [review]
Thanks for the documentation update :sparkles: 
I added a few notes inline :)

File: Docs/sphinx_documentation/source/GPU.rst (hunk 2 above)
Comment (mic84) [reply Docs/sphinx_documentation/source/GPU.rst]: Actually, it is still the case.

Comment (ax3l) [reply Docs/sphinx_documentation/source/GPU.rst]: Potentially better wording here as well (right terms):
```suggestion
**Note that AMReX supports NVIDIA GPU architectures with compute capability ``6.0`` or higher**.
```

> Actually, it is still the case.

Sorry, cannot follow. You mean "yes, we support CUDA 9.0+ and we should document that?" :)

Comment (maximumcats) [reply Docs/sphinx_documentation/source/GPU.rst]: NVIDIA

**Review by ax3l — APPROVED** [review]

**Pull request merged**